### PR TITLE
Migrate NestedIfDepth from AstNode.

### PR DESCRIPTION
**Pull request by lindamartin**

### CODE REVIEW

File: javascript-checks/src/main/java/org/sonar/javascript/checks/NestedControlFlowDepthCheck.java [hunk 1]
```diff
@@ -57,43 +65,94 @@ public int getMaximumNestingLevel() {
...
 
-  private static boolean isElseIf(AstNode astNode) {
-    return astNode.is(Kind.IF_STATEMENT) && astNode.getParent().is(Kind.ELSE_CLAUSE);
+  private void visitIf(IfStatementTree tree) {
+    scan(tree.condition());
+    scan(tree.statement());
+
+    ElseClauseTree elseClauseTree = tree.elseClause();
+    if (tree.elseClause() != null && elseClauseTree.statement().is(Kind.IF_STATEMENT)) {
+      visitIf(((IfStatementTreeImpl) tree.elseClause().statement()));
```
Comment (SonarTech) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/NestedControlFlowDepthCheck.java]: MAJOR Remove those useless parentheses. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessParenthesesCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

File: javascript-checks/src/main/java/org/sonar/javascript/checks/NestedControlFlowDepthCheck.java [hunk 2]
```diff
@@ -57,43 +65,94 @@ public int getMaximumNestingLevel() {
...
+      getContext().addIssue(this, tree,
+        String.format("Refactor this code to not nest more than %s if/for/while/switch/try statements.",
+          getMaximumNestingLevel()));
     }
   }
 
-  private static boolean isElseIf(AstNode astNode) {
-    return astNode.is(Kind.IF_STATEMENT) && astNode.getParent().is(Kind.ELSE_CLAUSE);
+  private void visitIf(IfStatementTree tree) {
+    scan(tree.condition());
```
Comment (benzonico) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/NestedControlFlowDepthCheck.java]: If you inline the call to the super.visitIfStatement() and handle the nestedLevel counter I guess that you don't have to check for else if or not and end up with simpler code.

Comment (lindamartin) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/NestedControlFlowDepthCheck.java]: After live discussion this could not be simplify.

Comment (benzonico) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/NestedControlFlowDepthCheck.java]: Yeap, my mistake :)

**Pull request merged**